### DISCUSSION TITLE
URGNENT: api/bunny Videos download failed

### DISCUSSION THREAD

**Discussion by shrewd-lab** [aclap-dev/video-downloadhelper#1145] [Help and Q&A]
### Your question

Hello,

I hope this message finds you well. I am encountering difficulties when trying to download videos where the video componenet url seems to be using the Bunny API. The error message I receive states:
---
Download failed

Sadly, we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!).
---
I have reported this issue multiple times, but I wanted to bring it to your attention here as well.
The #0 video component is as following: 	https://eXaMpLe.com/api/bunny/playlist/variant/eXaMpLe/course/eXaMpLe/1920x1080/video.m3u8?token=eXaMpLe&expires=1724198400

Thank you for your assistance, and I look forward to your help in resolving this matter.

Best regards,

### Addon & CoApp version and details

version: 9.1.0.41
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"a5b7d618218544f5"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "valid_license_message_has_been_displayed": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\Admin\\Documents"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 6
}

**Comment by Mattmjs** [comment]
Have you got an example if its publicly accessible,
if not could you supply another url as the posted url above expired.

**Reply by shrewd-lab** [reply]
I dont have an exmaple that's publically available.
The url i posted above was not real but was meant as an exmaple.
I manually reaplaced the original domain and the code with "example".
Is there anyway you can still help?
Are you the developer of videodownloadhelper?
did you get the issue i reported via the extension?

**Reply by GrampaWildWilly** [reply]
Neither Matt nor I is an employee of VDH.  We're both just users.

It would be helpful to post the `#0 main URL` from the Hit Details of one of your videos.  This time, don't censor the URL.  Sometimes, somebody who doesn't have an authorized user ID & password can get at content on a protected site if you post that URL.  Sometimes.  Not always.  But it's worth a shot.  Also, we would need to do it soon enough after you post it.  Your censored URL includes the qualifier `&expires=`, which says to me that your URLs have a shelf life.  I have no idea how long that shelf life might be, but again, it's worth a shot.

**Reply by shrewd-lab** [reply]
So sorry for the long wait.
Here is the Full URL withut censoring:

<html><body>

Tab | 1 (incognito: false)
-- | --
Marketing.MBA
https://marketing-mba.learningsuite.io/student/course/copywriting-de/ZFxs62DC/z9wTyKFZ/M9VmzvqC with 10 headers
https://storage.googleapis.com/learningsuite-prod-de-storage-cdn/cljr80b50112767901djgo0ve8m5%2Ftheme%2Fshared_link_preview_image.png?v=UfY3HN613nHrACIuz0lpHw==
true
https://api.learningsuite.io/api/bunny/playlist/master/Y2xqcjgwYjUwMTEyNzY3OTAxZGpnbzB2ZThtNTpjbHZ4dDIwdWYwc2trbDA0MzR6MG1uNW0z/course/3e0f8ec7-fedc-47da-a3fa-8def8749bb61/playlist.m3u8?token=B8F2SPEan09FSvw429bfaLQAbkeYe5VzfYUH-an_f5k&expires=1728691200
https://api.learningsuite.io/api/bunny/playlist/master/Y2xqcjgwYjUwMTEyNzY3OTAxZGpnbzB2ZThtNTpjbHZ4dDIwdWYwc2trbDA0MzR6MG1uNW0z/course/3e0f8ec7-fedc-47da-a3fa-8def8749bb61/playlist.m3u8?token=B8F2SPEan09FSvw429bfaLQAbkeYe5VzfYUH-an_f5k&expires=1728691200
none
Mp4 b:RawHls p:hls 0s
H264 at -1, fps: ?, q:?, no dim


</body>
</html>

**Reply by shrewd-lab** [reply]
I would also be up for providing login via private mail.

**Reply by GrampaWildWilly** [reply]
You actually posted 2 URLs there.  The first one prompts me for a login.  That tells me that at least it hadn't expired when I tried to use it.  But the login is blocking me.  The second one just gave me an error.

No, I refuse to accept logon credentials from anybody.  The VDH guys have repeatedly over the years, both in this forum & in our previous Google Groups forum, said that it is against company policy to do that.  I don't work for VDH but I take my cue from them.  I won't do it either.

**Reply by Mattmjs** [reply]
I didn't think about this in my initial post, can you check if you see this or not. 

Can you also share the media information again, I'm not sure if those m3u8 links expired 
or we're not permitted to access it as it gave an
error message. We don't need your login info, there are ways around that.

![371975942-e1a9cd45-4e3c-4702-ad81-21e76fe8f170](https://github.com/user-attachments/assets/c678ecb8-9d96-498a-9f79-e1943ccb88cd)

**Reply by shrewd-lab** [reply]
Hi,

Thanks for checking in again.

I am using firefox and not i dont see that, my url field looks like this: 
![image](https://github.com/user-attachments/assets/7388f8fb-d025-4202-8767-5245838702da)


And here is a new updated URL: 

https://api.learningsuite.io/api/bunny/playlist/variant/Y2xqcjgwYjUwMTEyNzY3OTAxZGpnbzB2ZThtNTpjbHZ4dDIwdWYwc2trbDA0MzR6MG1uNW0z/course/3e0f8ec7-fedc-47da-a3fa-8def8749bb61/1080p/video.m3u8?token=tocNNpFQlIzPIDBi8EQ0GHhm4vztunXvm8zDv3TWHYY&expires=1728777600

**Reply by Mattmjs** [reply]
Ok great so it means no DRM is used but there must be something else preventing the download. And your fresh new link won't
work as it still gives an error. It must only be connected to the user has access to it. You might need to share the network
activity and attach it here as .zip

To do that go to your video again and make sure the page has stopped loading, then open the developer tools : **Ctrl + Shift + E**.
This should open it with the Network selected. Reload the page as well as play the video for about 5 seconds.
After that follow the image to save it.


![3-10-23](https://github.com/user-attachments/assets/c77203a4-a169-404a-b5a1-99c5a6fc6bcf)

**Reply by shrewd-lab** [reply]
Hi, sorry for the late reply.
The zip file is over 25MB so here is a link to download it: https://we.tl/t-sG3bNGw7NF
I'm also adding the current URL: https://api.learningsuite.io/api/bunny/playlist/variant/Y2xqcjgwYjUwMTEyNzY3OTAxZGpnbzB2ZThtNTpjbHZ4dDIwdWYwc2trbDA0MzR6MG1uNW0z/course/3e0f8ec7-fedc-47da-a3fa-8def8749bb61/1080p/video.m3u8?token=S9sd7j0g4umx_kiMo0oRXgGwB_u5y6nwPGkGsPDVU3U&expires=1729296000

And here is a screenshot:
![image](https://github.com/user-attachments/assets/9c02be3c-c05e-444b-b2bc-0b0c5d66e442)


Thanks again for your help!

**Reply by Mattmjs** [reply]
Thanks, I went over it and it didn't offer any insights why it fails. I tried looking at the contents of the playlist/master url
but it shows as a long garbled string which is not what I want. And the playlist/master , playlist/variant urls produce an error
if I attempt to open them in a new tab. I opened one of the .ts segments which saved a 3 second piece. See the image below

I would in your position use those url's directly in ffmpeg or yt-dlp to see if there are any error messages or if it downloads your
video. I can't do that as the content uses a token and probably uses cookies since your the one accessing it.
Have you used these tools before ?

![api bunny](https://github.com/user-attachments/assets/70db84e3-5978-4bc0-bf74-37bd348eeaec)

**Reply by shrewd-lab** [reply]
Okay thank again for taking a look.
Really odd i've never faced any video that video-downloadhelper couldnt download over a browser.

I've downloaded ffmpeg in the past but had no luck for my inteded use (trying to bulk merge multiple mp4's into 1 single mp4 on my desktop).
But i get the feeling it's out of my skill level using this command prompt tools. I guess i'll have to give up here. :-(

**Reply by GrampaWildWilly** [reply]
When you say "bulk merge" are you trying to string together videos one after the other into a single video?  If so, take a look at the concat parameter of ffmpeg.  I know the ffmpeg documentation is more like something you would read when you actually already know what it says.  But the concat parameter is explained well enough that I figured it out.  The ffmpeg documentation is packaged with ffmpeg as a directory of HTML files, meaning web pages to read in a browser.  Try again.  Concatenating videos together is pretty easy.  The command I use looks like this:

ffmpeg -f concat -safe 0 -i "input list file" -codec: copy "output file name" 2>"output log file"

Surround each file name with quotation marks.

The input list file is a collection of lines that look like this:

file 'file name'

It's the word file followed by the name of a file.  Each file name must be enclosed in apostrophes, not quotation marks.  The order of lines in this file determines the order of "welding together" of the input videos in the output file.  All the file names must be fully qualified, meaning of the form x:\directory\directory\directory\file name.  That goes for the file names in the input list file as well as on the ffmpeg command itself.

When the command completes, just read the output log file in your text editor to see what happened, verify things went the way you wanted, see what went wrong so you can try again.
